soh/soh/Enhancements/crowd-control/CrowdControl.cpp

```diff
@@ -144,8 +141,21 @@ void CrowdControl::ListenToServer() {
 
             // If effect is a one off run, let's execute
             if (!incomingEffect->timeRemaining) {
-                EffectResult result =
-                ExecuteEffect(incomingEffect->type.c_str(), incomingEffect->value, false);
+                EffectResult result = EffectResult::Retry;
+                if (incomingEffect->category == "spawn_enemy") {
+                    if (GameInteractor::CanSpawnEnemy()) {
+                        if (GameInteractor::Actions::SpawnEnemyWithOffset(incomingEffect->value[0],
+                            incomingEffect->value[1])) {
+                            result = EffectResult::Success;
+                        }
+                    }
+                } else {
+                    result = CrowdControl::CanApplyEffect(incomingEffect);
+                    if (result == EffectResult::Success) {
+                        GameInteractor::ApplyEffect(incomingEffect->giEffect);
+                    }
```

Comment:
similar to the comment about enemy spawning, it'd be nice to move the check into `ApplyEffect`

since this is all cpp, we can even add a bool to skip the `CanApply` check with a default value of false to `ApplyEffect` (and do a similar thing for enemy spawns) if we want to allow skipping that check